Lab1, Есин Никита

KazuruK opened the pull request.

KazuruK committed:
# Lab1: Console Chat v1.3
## inital pull request commit

KazuruK committed:
Update README.md

h31 commented:
Протестировать на сервере + мне посмотреть код

h31 commented on server.py:
```diff
@@ -0,0 +1,94 @@
+    print(f'Listening for connections on {IP}:{PORT}...')
+    while True:
+        client_socket, client_address = server.accept()
+        user = receive_message(client_socket)
```
К сожалению, не помню, показывал ли вам эксперимент с `nc`. Если вдруг не показывал, напишите мне или спросите у Александра.

h31 commented on server.py:
```diff
@@ -0,0 +1,94 @@
+            return False
+        message_length = int(message_header.decode(ENCODING))
+        return {'header': message_header, 'data': sock.recv(message_length)}
+    except Exception as e:
```
`as e`?

h31 commented on server.py:
```diff
@@ -0,0 +1,94 @@
+HEADER = int(os.getenv('HEADER'))
+IP = os.getenv('IP')
+PORT = int(os.getenv('PORT'))
+ENCODING = os.getenv('ENCODING')
```
Зачем?

h31 commented on server.py:
```diff
@@ -0,0 +1,94 @@
+        if not len(message_header):
+            return False
+        message_length = int(message_header.decode(ENCODING))
+        return {'header': message_header, 'data': sock.recv(message_length)}
```
Проверьте, работает ли этот код в реальной сети.

KazuruK committed:
# Lab1: Console Chat v1.3
## real network test commit

KazuruK committed:
Merge remote-tracking branch 'origin/esinnikita/1lab' into esinnikita/1lab

KazuruK committed:
Edit receive methods from client and server for remoete file-sending

KazuruK committed:
Добавил в receive_message while для правильной обработки передачи файлов. Столкнулся с ошибкой ключей.

KazuruK committed:
# receive_message
    - Реализовано чтение частями при файле большем 10000бт
    - Добавлена задержка 0.1с при чтении файла меншем 10000бт
    - Добавлена защита от вредоносных клиентов при подключении к серверу.

Клиент-серверное взаимодействие протестированно на удаленном сервере в режиме удаленной отладки.

KazuruK committed:
# receive_message
    - Удалена здаержка 0.1 в определенных ситуациях
    - Суммирование сообщений переделано под .append()

# user_connect
    - Часть работы main() перенасена в user_connect для вызова в потоке. Это решает проблему вредоносных клиентов.

Клиент-серверное взаимодействие протестированно на удаленном сервере в режиме удаленной отладки.
